core/src/main/java/edu/wpi/grip/core/settings/ProjectSettings.java

```diff
@@ -134,6 +160,56 @@ private String computeFRCAddress(int teamNumber) {
     return "roboRIO-" + teamNumber + "-FRC.local";
   }
 
+  public void setPreferredGeneratedLanguage(String lang) {
+    checkArgument(lang.matches("Java|C\\+\\+|Python"),
```

Comment:
We have an enum in code gen to represent this already. Why not use that to convert/check if the input is valid.